### PR TITLE
Add note about ExceptionHandler in CLI

### PR DESCRIPTION
**Pull request by hkdobrev**

### CODE REVIEW

File: doc/cookbook/error_handler.rst [hunk 1]
```diff
@@ -36,3 +36,18 @@ the ``$debug`` argument::
     use Symfony\Component\Debug\ExceptionHandler;
 
     ExceptionHandler::register(false);
+
+
```
Comment (fabpot) [inline doc/cookbook/error_handler.rst]: extra blank line

File: doc/cookbook/error_handler.rst [hunk 2]
```diff
@@ -36,3 +36,18 @@ the ``$debug`` argument::
...
+.. note::
+
+    Important caveat when using Silex in a command-line interface:
+    The ``ExceptionHandler`` should not be enabled as it would convert an error
+    to HTML output and return a non-zero exit code.
+
+    .. code-block:: php
```
Comment (fabpot) [inline doc/cookbook/error_handler.rst]: Should be removed in favor of replacing the `.` at the end of the previous sentence by `::`

File: doc/cookbook/error_handler.rst [hunk 3]
```diff
@@ -36,3 +36,18 @@ the ``$debug`` argument::
     use Symfony\Component\Debug\ExceptionHandler;
 
     ExceptionHandler::register(false);
+
+
+.. note::
+
+    Important caveat when using Silex in a command-line interface:
```
Comment (fabpot) [inline doc/cookbook/error_handler.rst]: on a command-line interface?

File: doc/cookbook/error_handler.rst [hunk 4]
```diff
@@ -36,3 +36,18 @@ the ``$debug`` argument::
...
+    The ``ExceptionHandler`` should not be enabled as it would convert an error
+    to HTML output and return a non-zero exit code.
+
+    .. code-block:: php
+
+        use Symfony\Component\Debug\ExceptionHandler;
+
+        if (!in_array(PHP_SAPI, ['cli', 'phpdbg'])) {
```
Comment (xabbuh) [inline doc/cookbook/error_handler.rst]: What about `cli-server` here?

Comment (hkdobrev) [reply doc/cookbook/error_handler.rst]: @xabbuh If you mean the built-in PHP server meant for development, it shouldn't be included here as it's alright converting error to HTML in that SAPI.

**Comment by hkdobrev** [comment]
Created https://github.com/symfony/symfony/pull/26677 so `Debug::enable()` supports PHPDBG as well.

**Comment by hkdobrev** [comment]
I'm happy to change the whole document to use just `Debug::enable()` as long as https://github.com/symfony/symfony/pull/26677 is released.

I'm trying to get this CLI-specific changes in as we had a minor outage on production, because of this issue as errors were not caught properly in CLI and tests.

**Comment by hkdobrev** [comment]
I've addressed all inline comments. Let me know if all should be replaced with `Debug::enable()` or that should be separate PR. Thanks!

**Comment by fabpot** [comment]
Thank you @hkdobrev.

**Pull request merged**